### ISSUE TITLE
Netflix support in SteamOS BPM?

### ISSUE THREAD

**Issue by 3vi1** [ValveSoftware/SteamOS#417]
I've seen people report that Netflix now works natively in the SteamOS browser, but on my system (Alienware i7 Steam Machine, opted into brewmaster_beta, w/ beta steam client) I still see the "Whoops.." error and Netflix refuses to actually stream anything if I try to use it from the browser in SteamOS's normal BPM.

If I go to desktop mode and fire up Chrome, I can watch Netflix fine (so it's definitely not a network issue).

Is Netflix supposed to be working in BPM, or are all the people saying it works really talking about steam in BPM on Windows instead of SteamOS?

**Comment by egee-irl** [comment]
Can you confirm you are using SteamOS Brewmaster and *not* SteamOS Alchemist? I watch Netflix on my Steam Machine every night using the SteamOS browser but I am running Brewmaster.

**Comment by 3vi1** [comment]
I'm definitely using Brewmaster_beta.  The apt sources say brewmaster in sources.list, and brewmaster_beta in steamos-beta-repo.list.  The Alienware steam machines shipped with brewmaster, so I've never even had Alchemist on the box.

> desktop@steamos:~$ lsb_release -a
> No LSB modules are available.
> Distributor ID: SteamOS
> Description:    SteamOS 2.0
> Release:        2.0
> Codename:       brewmaster

It's puzzling why it works for you and others, yet I always get error M7355-1204.  Other sites like Yahoo stream fine via the SteamOS browser.

I've tried clearing the browser cache via system setting, but the symptoms have not changed.

![image](https://cloud.githubusercontent.com/assets/628806/11309842/d608f628-8f8b-11e5-87c0-9900558c3bdb.png)

Thanks for taking the time to help me troubleshoot this.  Any other ideas are appreciated.

![image](https://cloud.githubusercontent.com/assets/628806/11309271/8504225a-8f88-11e5-8e4c-ca6146669d68.png)

**Comment by 3vi1** [comment]
BTW:  Googling for the error, the closest match I saw was M7355-1203.  That error indicates an issue with the browser, and their suggested fix is to re-install the browser.  Not something I can really do without knowing a little more about the way the SteamOS components are grouped.  I looked via aptitude and did not see a separate package for the integrated web browser .  :\

**Comment by ProfessorKaos64** [comment]
The only thing I can think of is chopping up the directory that has the steam client, forcing it to redownload. It's messy, and I don't have the specifics on hand. Essentially you are forcing Steam to redownload the client components. Have you tried a restore? This may will redownload the latest client and maybe undo this kink. However, it _will_ wipe out any custom installed software you might have added.

**Comment by 3vi1** [comment]
I was hoping to avoid the restore option.  I've built some emulators for my old games and other software from source and really want to avoid all the reconfiguration I'd inevitably have to do when I forget to back-up some config.

I would have thought opting in/out/in of the client beta would have rewritten the files related to the integrated web browser, since that basically re-installs the client.  I've been looking for any local cache for it that I could manually clear, but haven't seen anything promising yet.

**Comment by ProfessorKaos64** [comment]
>  I've built some emulators for my old games and other software from source

Which ones? I may already have them packaged for you, if not you can request them over at the SteamOS-Tools GitHub page.  I know this sucks, just throwing out suggestions for you. Have you posted this to on the Steam Universe discussion board? How about /r/SteamOS and /r/linux_gaming? There is also a dedicated subreddit called /r/steambox.

**Comment by 3vi1** [comment]
> Which ones

Mess is the main one - saves me from having to have the old NES, Genesis, and other systems hooked up.  The real pain would be recreating the shortcuts for each user, since they have the long parameters set in the launch options.  Then again, I recall finding the file that contains those and backing it up on my Linux system long ago, so I could probably find it again.

I prefer to build these from source rather than use a package, since I like to go in and do things like disable the info screens and such.

**Comment by egee-irl** [comment]
> I've tried clearing the browser cache via system setting, but the symptoms have not changed.

I am curious if there is a way to check the specific version of the browser. Seems like the problem is there..

From what I understand, Error M7355-1203 happens not because the browser doesn't support Netflix, but because Netflix doesn't support *something* about the browser. An example of this is before Google updated Chrome to handle DRM within HTML5 videos. You'd see this error because Netflix wouldn't allow the browser to view the videos because it was lacking the DRM plugins, or whatever.

**Comment by 3vi1** [comment]
Yeah, I was one of the early pipelight users on Linux, and jumped ship as soon as HTML5 worked (with forged useragent)...  I went through my Netflix settings to make sure there wasn't still any settings relevant to that, but found nothing.  :\

**Comment by 3vi1** [comment]
I fixed it... though I'm not entirely sure how.

Yesterday I tried going through all of /home/steam looking for anything related to caching and blowing it away (well, renaming it at least <g>) indescriminately.  However, the problem still persisted.

Just now, I ran clonezilla and backed up sda in preparation for a complete re-install.  But then, on a lark, I decided to try to clear the web browser cookies and opt-out of the beta again.  I did, tested, and Netflix began playing correctly!  I opted back into the beta, and Netflix still appears to be working.

So, I'm not sure what the specific fix was - maybe a mix of clearing out some of the cached html and then re-installing the client by flip-flopping my beta status.  But whatever the original cause, I no longer have the issue.

Thanks to egee-irl and prof K for their suggestions and help in determining that the issue was specific to my install!

**Comment by pic1701d** [comment]
I'm having the same issue here.
I did a fresh install today, opted in brewmaster_beta and steam client beta.
Now if I want to watch a movie I get the "Hoppla.." error (which seems to be the german translation for "Whoops" ;-) )

How exactly did you clean the cached html and the cookies. I did not find anything like that under  /home/steam?

**Comment by retuxx** [comment]
I'm having exactly the same issue using a custom SteamOS installation.

**Comment by ProfessorKaos64** [comment]
Those of you still affected could try clearing your browser cache at `/home/steam/.local/share/Steam/config/htmlcache`  I have no been affected by this issue, so please backup the folder before giving this a try.

**Comment by retuxx** [comment]
I used the following commands

```bash
cd /home/steam/.local/share/Steam/config/ # change into the directory containing htmlcache
sudo tar cfvz htmlchache.tar.gz htmlcache # create backup
sudo rm -rf htmlcache/* # remove all files
sudo shutdown -r now # reboot system
```

Still having the same problem :(

**Comment by pic1701d** [comment]
Since this happens even after a fresh install without any modifications, maybe the issue should be reopened?

**Comment by 3vi1** [comment]
Re-opening, since it appears to be affecting more than one other person and the cause/fix has yet to be identified.

**Comment by retuxx** [comment]
I've installed Google Chrome to find out whether this is a SteamOS or a Steam-Client related bug. As expected, using Chrome to watch Netflix works fine. After installing Chrome and testing Netflix I tried the Steam-Browser again (Maybe Chrome installed packages that are required for Steam to handle Netflix) but no change here.

**Comment by retuxx** [comment]
Using the newest version of Steam fixes the problem for me.

**Comment by mohsinhasan** [comment]
I did a fresh install of SteamOS a couple of days ago and am on brewmaster-beta. I am also having the same issue. Netflix can't run on the default browser with the "Whoops.." error.

Is this issue actually resolved?

**Comment by Plagman** [comment]
Go watch a community broadcast in Steam, then it should start working.

**Comment by pic1701d** [comment]
That worked for me. Thanks!

**Comment by ProfessorKaos64** [comment]
I have added this to the wiki under troubleshooting [here](https://github.com/ValveSoftware/SteamOS/wiki/Troubleshooting#i-cannot-watch-netflix-or-amazon-web-content)

**Comment by mohsinhasan** [comment]
@Plagman That fixed it. Thanks for the tip!

Now on to the next problem. Although I can see Netflix now, it does stutter quite a bit and the image quality doesn't appear to be full res. It's probably decoding this on the CPU vs using GPU decoding. Playing Netflix on Chrome through desktop mode runs at full speed and full res.

Is there any way I can enable GPU decoding on the BPM browser?

**Comment by Enigma0** [comment]
Seconding basic functionality works but poor performance/playback/stuttering.

**Comment by ProfessorKaos64** [comment]
@Enigma0 , what is your network setup? your provider and speed? I use this everyday, no issues. 30 Mb/s down cable over wireless N.

**Comment by Enigma0** [comment]
Hardware:

- PC with 660 ti
- 6 core amd cpu @ 3.5ghz+
- 8gb ram
- 1tb hdd

Network:

- 1gig lan through switch to router
- router does n/ac 2.4ghz/5ghz @300mbps+
- 150/150 mbps ISP
